Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,302 @@
+import edu.asu.conceptpower.servlet.exceptions.LuceneException;
+import edu.asu.conceptpower.servlet.lucene.ILuceneUtility;
+import edu.asu.conceptpower.servlet.reflect.LuceneField;
+import edu.asu.conceptpower.servlet.reflect.SearchField;
+import edu.asu.conceptpower.servlet.rest.LuceneFieldNames;
+import edu.asu.conceptpower.servlet.wordnet.WordNetConfiguration;
+import edu.mit.jwi.Dictionary;
+import edu.mit.jwi.IDictionary;
+import edu.mit.jwi.item.IIndexWord;
+import edu.mit.jwi.item.ISynset;
+import edu.mit.jwi.item.IWord;
+import edu.mit.jwi.item.IWordID;
+import edu.mit.jwi.item.POS;
+
+@Component
+@PropertySource("classpath:lucene.properties")
+public class LuceneUtility implements ILuceneUtility {
+
+    @Autowired
+    private WhitespaceAnalyzer whiteSpaceAnalyzer;
+
+    @Autowired
+    private WordNetConfiguration configuration;
+
+    @Value("${lucenePath}")
+    private String lucenePath;
+
+    private IndexWriter writer = null;
+    private IndexReader reader = null;
+
+    @PostConstruct
+    public void init() throws LuceneException {
+        try {
+            Path relativePath = FileSystems.getDefault().getPath(lucenePath, "index");
+            Directory index = FSDirectory.open(relativePath);
+            IndexWriterConfig configWhiteSpace = new IndexWriterConfig(whiteSpaceAnalyzer);
+            writer = new IndexWriter(index, configWhiteSpace);
+            reader = DirectoryReader.open(writer, true);
+        } catch (IOException ex) {
+            throw new LuceneException("Restart the application", ex);
+        }
+    }
+
+    public void deleteById(String id) throws LuceneException {
+        try {
+            Query q = new QueryParser("id", whiteSpaceAnalyzer).parse("id:" + id);
+            writer.deleteDocuments(q);
+            writer.commit();
+        } catch (IOException ex) {
+            throw new LuceneException("Issues in deletion. Please retry", ex);
+        } catch (ParseException e) {
+            throw new LuceneException("Issues in framing queries", e);
+        }
+
+    }
+
+    public void insertConcept(ConceptEntry entry) throws LuceneException, IllegalAccessException {
+        Document doc = new Document();
+
+        java.lang.reflect.Field[] fields = entry.getClass().getDeclaredFields();
+        for (java.lang.reflect.Field field : fields) {
+
+            LuceneField searchFieldAnnotation = field.getAnnotation(LuceneField.class);
+            field.setAccessible(true);
+            if (searchFieldAnnotation != null) {
+
+                try {
+                    Object contentOfField = field.get(entry);
+                    if (contentOfField != null) {
+                        doc.add(new StringField(searchFieldAnnotation.lucenefieldName(),
+                                String.valueOf(contentOfField).replace(" ", ""), Field.Store.YES));
+                    }
+
+                } catch (IllegalArgumentException ie) {
+                    throw new IllegalArgumentException(ie);
```

Comment:
either create a new exception with a custom exception class or throw the existing one, but this doesn't make sense.